aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -22,12 +22,41 @@ using namespace at::sparse;
 
 namespace {
 
-  inline SparseTensor get_result_tensor_for_unary_op(const SparseTensor& input) {
-    if (c10::isIntegralType(input.scalar_type(), /*includeBool=*/true)) {
+  inline SparseTensor get_result_tensor_for_unary_op(const SparseTensor& input, bool promotes_integer_to_float = false) {
+    if (c10::isIntegralType(input.scalar_type(), /*includeBool=*/true) && promotes_integer_to_float) {
       return at::empty_like(input, input.options().dtype(c10::get_default_dtype()));
     }
     return at::empty_like(input);
   }
+
+  // --------------------------------------------------------------------
+  // helper function for sanity check of input and output tensors
+  // this helper function is for unary ops
+  // --------------------------------------------------------------------
+  static inline void sparse_unary_op_check_sanity(SparseTensor& result, const SparseTensor& self,
```

Comment:
This is a cool function, but it does more than "sanity" check because it can copy self into result. Maybe there's a better name for it? Also, the comment should reflect the property it guarantees (that the result tensor is a coalesced copy of the input (self) tensor).

Reply:
Suggestion: `sparse_unary_op_check_and_init` where `init` refers to copying `self` to `result` (unless these tensors are the same).